documentation/modules/oauth/con-oauth-authorization-keycloak-authorization-services.adoc

```diff
@@ -0,0 +1,199 @@
+// Module included in the following module:
+//
+// proc-oauth-authorization-broker-config.adoc
+
+[id='con-oauth-authorization-keycloak-authorization-services_{context}']
+= Manage policies and permissions in Keycloak Authorization Services
+
+== Kafka Authorization Model
+
+The link:https://kafka.apache.org/documentation/#security_authz_primitives[Kafka authorization model] defines resource types, and permissions available for each type.
+Whenever some action is performed by the Kafka client, one or more permissions are checked by the broker (depending on the action performed) using the configured authorizer.
+
+There are five resource types: `Topic`, `Group`, `Cluster`, `TransactionalId`, `DelegationToken`.
+
+And these are the different permissions for each resource type:
+
+Topic:
+
+* Create
+* Write
+* Read
+* Delete
+* Describe
+* DescribeConfigs
+* Alter
+* AlterConfigs
+
+Group:
+
+* Read
+* Describe
+* Delete
+
+Cluster:
+
+*  Create
+*  Describe
+*  Alter
+*  DescribeConfigs
+*  AlterConfigs
+*  IdempotentWrite
+*  ClusterAction
+
+TransactionalId:
+
+*  Describe
+*  Write
+
+DelegationToken:
+
+* Describe
+
+
+== Keycloak Authorization Services Model
+
+There are four concepts used in Keycloak Authorization Services to grant permissions: `resources`, `authorization scopes`, `policies`, and `permissions`.
+
+`Resources` are a set of resource definitions that are used to match permitted actions.
+For example, a resource can be an individual topic, or it can be a set of all topics with names that start with some prefix.
+The resource definition has a set of available `authorization scopes` associated with it. These represent a set of all actions that can be permitted.
```

Comment:
```suggestion
The resource definition has a set of available authorization scopes associated with it., which represent a set of all actions that are permitted.
```

Reply:
In fact the original meaning 'can be permitted' is different than the suggested meaning 'are permitted'. The idea here is not to give permissions by adding the to list, but to compose a list of available permissions for this particular resource instance. Then on the Policies tab the granting of permissions on this particualr resource instance will be limited to 'can be permitted' actions. Maybe:
```suggestion
The resource definition has a set of available authorization scopes associated with it, which represent a set of all actions available on the particular resource.
```